%YAML:1.0

#--------------------------------------------------------------------------------------------
# ROS topics
#--------------------------------------------------------------------------------------------
imuTopic: "/imu0"
leftImageTopic: "/cam0/image_raw"

readFromRosBag: 1   # 1: read from rosbag; 0: subscribe from rosbag (you should 'rosbag play ...')
# note: 'rosBag' is configurated in launch file

#--------------------------------------------------------------------------------------------
# Paramters!
#--------------------------------------------------------------------------------------------
downSampleRate: 1  # default: 1
KeyPointNumber: 200
ThresholdOfPredictNewKeyPoint: 1.0
HalfPatchSize: 5

# You can load keypoints detected by other methods.
# In this case, a corresponds.txt file should be provided to indicate the
# correspondences between timestamp and filename
LoadDetectedKeypoints: 0
DetectedKeypointsFile: "/../data/SuperPoints/V1_03_difficult"


#--------------------------------------------------------------------------------------------
# Camera Parameters. Adjust them!
#--------------------------------------------------------------------------------------------
Camera.type: "PinHole"

# Camera calibration and distortion parameters (OpenCV) 
Camera.fx: 458.654
Camera.fy: 457.296
Camera.cx: 367.215
Camera.cy: 248.375

Camera.k1: -0.28340811
Camera.k2: 0.07395907
Camera.p1: 0.00019359
Camera.p2: 1.76187114e-05

# Camera resolution
Camera.width: 752
Camera.height: 480

# Camera frames per second 
Camera.fps: 20

# Color order of the images (0: BGR, 1: RGB. It is ignored if images are grayscale)
Camera.RGB: 1

# Transformation from camera to body-frame (imu)
Tbc:
    [0.0148655429818, -0.999880929698, 0.00414029679422, -0.0216401454975,
     0.999557249008, 0.0149672133247, 0.025715529948, -0.064676986768,
    -0.0257744366974, 0.00375618835797, 0.999660727178, 0.00981073058949,
     0.0, 0.0, 0.0, 1.0]

# IMU noise
IMU.NoiseGyro: 1.7e-4       # rad/s/sqrt(Hz)
IMU.NoiseAcc: 2.0000e-3     # m/s^2/sqrt(Hz)
IMU.GyroWalk: 1.9393e-05    # rad/s^2/sqrt(Hz)
IMU.AccWalk: 3.0000e-03     # m/s^3/sqrt(Hz)
IMU.Frequency: 200
